### PR TITLE
feat(rgbpp-sdk/ckb): Unlock all rgbpp asset of the same rgbpp lock

### PR DESCRIPTION
**Pull request by duanyytop**
## Main Changes

- Unlock all RGB++ assets(the same xudt type) of the same RGBPP lock script

### CODE REVIEW

**Comment by duanyytop** [comment]
BTC TX:  https://mempool.space/testnet/tx/0f54c6ed3b644515bae3f50825d1d5168f5e3a842988c972c3c5ed635aeaf486
CKB TX: https://pudge.explorer.nervos.org/transaction/0xb4aeecc36b622f0d1945d6b9cc34833759349526038f0dadca1250d383fdda5a

**Comment by Flouse** [comment]
Suggestion:

When there are change sats to be returned, don't return them to RGB++ UTXO.

**Comment by duanyytop** [comment]
For the situation where there are both RGB++ and BTC changes, it is recommended according to the following transaction structure：

```yaml
# BTC_TX
input:
	btc_vin_1
        btc_vin_2
	...
output:
	OP_RETURN: commitment
	btc_utxo_1 # for the receiver RGB++ assets (dust satoshi)
	btc_utxo_2 # for the sender RGB++ change (dust satoshi)
	btc_utxo_3 # for the sender BTC change 
```

**Review by Flouse — APPROVED** [review]

File: packages/ckb/src/collector/index.ts [hunk 1]
```diff
@@ -124,7 +132,7 @@ export class Collector {
       });
       sumInputsCapacity = sumInputsCapacity + BigInt(cell.output.capacity);
       sumAmount += leToU128(cell.outputData);
-      if (sumAmount >= needAmount) {
+      if (sumAmount >= needAmount && !isMax) {
```
[the remark this answers is not in the document]
Comment (duanyytop) [reply packages/ckb/src/collector/index.ts:135]: Well, this will indeed be clearer, I will update it in the PR later

**Pull request merged**